graphbolt/src/feature_cache.cc

```diff
@@ -0,0 +1,152 @@
+/**
+ *   Copyright (c) 2023, GT-TDAlab (Muhammed Fatih Balin & Umit V. Catalyurek)
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ * @file feature_cache.cc
+ * @brief Feature cache implementation on the CPU.
+ */
+#include "./feature_cache.h"
+
+namespace graphbolt {
+namespace storage {
+
+constexpr int kIntGrainSize = 64;
+
+S3FifoCachePolicy::S3FifoCachePolicy(int64_t capacity)
+    : S_{capacity / 10},
+      M_{capacity - capacity / 10},
+      position_q_{0},
+      ghost_q_time_{0},
+      capacity_{capacity} {}
```

Comment:
I have been using this syntax because it is the new default best way to initialize variables in modern C++: https://en.cppreference.com/w/cpp/language/initialization

However to follow the guidelines, I am changing to using paranthesis.